pipe.go

```diff
@@ -240,6 +245,22 @@ func DialPipeAccess(ctx context.Context, path string, access uint32) (net.Conn,
 		return nil, err
 	}
 
+	info, err := windows.GetSecurityInfo(windows.Handle(h), windows.SE_FILE_OBJECT, windows.OWNER_SECURITY_INFORMATION)
```

Comment:
If this Owner check is good enough, then I'll +1 the PR.

Out of curiosity I looked a bit further if we still wanted to compare DACL. Last session we tried casting pointers with unsafe, but it looks like we can use the `sys/windows` package. It provides the security_descriptor.DACL() method https://cs.opensource.google/go/x/sys/+/refs/tags/v0.6.0:windows/security_windows.go;l=1188. And that DACL() method goes and makes another syscall using advapi32.dll:
https://cs.opensource.google/go/x/sys/+/master:windows/zsyscall_windows.go;l=799?q=getSecurityDescriptorDacl&ss=go%2Fx%2Fsys which should get us the DACL as a Go struct.